src/sql/sqlops.proposed.d.ts

```diff
@@ -1146,5 +1146,12 @@ declare module 'sqlops' {
 		 * @param connectionId The ID of the connection
 		 */
 		export function getUriForConnection(connectionId: string): Thenable<string>;
+
+		/**
+		 * Opens the connection dialog, calls the callback with the result. If connection was successful
+		 * returns the connection otherwise returns undefined
+		 * @param callback
+		 */
+		export function openConnectionDialog(provider: string[], callback: (connection: connection.Connection) => void);
```

Comment:
This should work but maybe it would make more sense to do something like you did with the deferred promise above and this could just be a thenable? It would probably make extension code a bit easier to write since they could just await this call then.